exaudfclient/base/python/python3/python_ext_dataframe.cc

```diff
@@ -394,13 +446,20 @@ inline bool isNumpyDatetime64(const char* typeName){
     return std::string(typeName).find("datetime64[")==0;
 }
 
+inline bool isArrowDecimal128(const char* typeName){
+    // example decimal128(3, 2)[pyarrow]
+    return std::string(typeName).find("decimal128(")==0 && std::string(typeName).find("[pyarrow]")!=std::string::npos;
```

Comment:
1. You are instantiating 2x the same std::string. It could happen that you have 2 heap allocations here.
2. For the first search you just want to check for the prefix, so you don't need a whole string search which is expansive. Check out https://stackoverflow.com/a/40441240 or use strncmp().
3. Same for the second where you want to compare the suffix. Checkout https://stackoverflow.com/a/2072890 or use strncmp().

Reply:
I will take a look, also this function doesn't get too often called, only once per dataframe, so we don't need to optimize too hard.